vsm

```diff
@@ -165,6 +167,15 @@ class State(object):
                     item = item.replace(" ", "").split("=")
                     vars(self.variables)[item[0]] = item[1]
 
+        # inject this object into the globals dictionary so it will be available
+        # to the function we're executing (since it won't really be filled in
+        # until after this constructor completes)
+        global_vars = globals()
+        global_vars["state"] = self
```

Comment:
This is because it's ultimately being accessed by the `emit()` function but in an indirect way; the call to `emit()` is getting parsed from a string so we have to make the `State` class instance (the `state` global variable) global so it will always be accessible by `emit()`. We manually add it to the globals dict here as a special case because this call to `emit()` happens (just) before the `State` class instance is finished being constructed and thus before the assignment to the global `state`.